### PR TITLE
fix: add migrate stage to backend docker

### PR DESCRIPTION
**Pull request by chriscasto**
# Pull Request Template

## Issue Overview

This PR addresses #59 
- [X] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

The docker images generated for the backend service lack some dependencies for executing `yarn db:*` commands, such as the postgres client.  This PR adds these dependencies into a separate build stage named "migrate" that builds upon the "run" target.  This will keep the runtime image small, improving build and startup times, and will keep each image focused on a single task in accordance with best practices.

## How Can This Be Tested/Reviewed?

From the repo root, build the image: 
`docker build --file build/docker/Dockerfile.backend --target migrate -t doorway/backend:migrate ./backend/core`

Then reseed your database with the generated image (below assumes default user/password/etc):
`docker run --env-file backend/core/.env -e PGUSER=bloom-dev -e PGPASSWORD=bloom -e PGHOST=localhost -e PGDATABASE=bloom --network host doorway/backend:migrate yarn db:reseed`

If you get an error like this
image

Drop the database first with the user you created it with by running `yarn db:drop` from the `./backend/core` directory, then `cd` back to repo root and run the above command again.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [X] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [X] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: build/docker/Dockerfile.backend
Comment (isaac-looker) [inline build/docker/Dockerfile.backend:111]: do we want to lock on a specific version of postgres?

**Review by isaac-looker — APPROVED** [review]

File: build/docker/Dockerfile.backend
Comment (chriscasto) [reply build/docker/Dockerfile.backend:111]: I don't think it will matter too much given that this is the client and not the server.  The only thing the client is used for is for some DDL commands that set the application up for the real interaction with the server.

**Comment by chriscasto** [comment]
@phattya Can you take a look at this when you get the chance?

**Review by phattya — APPROVED** [review]
Approving to unblock, tho we ran into psql permissions issues. It runs on AWS for Chris, so good enough for now.

**Comment by chriscasto** [comment]
Just a note before merging that there were some testing errors that popped up for reviewers when using the instructions provided.

When `psql` tries to connect to the server, it needs to know which database to use.  That's what `PGDATABASE` is for.  The problem is that if the database doesn't exist, it can't connect to it ("database 'bloom' does not exist"), but if it does exist, you can't drop it ("cannot drop the currently open database").  The reason why this worked when running on AWS is because the database user there has its own database that it connects to before dropping/creating the "bloom" table.  I'm still not sure why this worked for me locally.

The revised test instructions are as follows:

**Create the bloom database (just in case)**
`yarn db:create`

**Create a bloom-dev database**
`sudo docker run --env-file backend/core/.env -e PGUSER=bloom-dev -e PGPASSWORD=bloom -e PGHOST=localhost -e PGDATABASE=bloom --network host doorway/backend:migrate psql -c 'CREATE DATABASE "bloom-dev";'`

**Drop the bloom database (to reset ownership)**
`yarn db:drop`

**Seed the data (note the absence of PGDATABASE)**
`sudo docker run --env-file backend/core/.env -e PGUSER=bloom-dev -e PGPASSWORD=bloom -e PGHOST=localhost --network host doorway/backend:migrate yarn db:reseed`

**Comment by phattya** [comment]
FYI...Chris was able to provide updated instructions that worked for testing.

`sudo docker run --env-file backend/core/.env -e PGUSER=bloom-dev -e PGPASSWORD=bloom -e PGHOST=localhost -e PGDATABASE=bloom --network host doorway/backend:migrate psql -c 'CREATE DATABASE "bloom-dev";'`

`yarn db:drop`

`sudo docker run --env-file backend/core/.env -e PGUSER=bloom-dev -e PGPASSWORD=bloom -e PGHOST=localhost --network host doorway/backend:migrate yarn db:reseed`

These 3 commands completed successfully. Fully tested & approved now.

**Pull request merged**
